[DPP-1231] Polish Unique Select styles

sabs231 opened the pull request:
## What did you implement:

Implementing issue: [DPP-1231](https://fng-jira.fox.com/browse/DPP-1231)

## How did you implement it:
Update the styles for a better user experience

## Files to review:
Please review the following files and ignore the rest because they are auto generated by `nps`
1. `less/control.less`
1. `less/select.less`
1. `src/Async.js`
1. `src/Select.js`
1. `src/utils/defaultArrowRenderer.js`


## How can we verify it:
1. In the DPP repo `fng-cms` change branch to `chore/DPP-1231`
1. On the DPP `package.json` file replace the `@foxdcg/react-select` dependency with this:

```json
"@foxdcg/react-select": "git://github.com/foxbroadcasting/react-select.git#chore/DPP-1231",
```

1. Run `npm i && npm start`
1. Open the DPP on the browser
1. Go to `Types` section
1. Change branch to `dev-chore/DPP-1231`
1. Click on the `+` to open Global Creation flow
1. Select `New Query`
1. Select `Video Queries`
1. Select `None` (it can be any type of Query)
1. Select a panel from the `Assign To Panel` input on the top right corner of the Create a new Video Query screen
1. Search for `FX Presents` and select it
1. Click outside of the Select component and again on the select component
1. Look for `FX Presents` again
1. Notice the selected option is aligned as in the gallery [mocks](https://gallery.io/projects/MCHbtQVoQ2HCZfNvzPdzBSok/files/MCHtA7U1iMGr67g5L1tTjT-XhkieAxJkg6o)
1. Profit 💰 

## Screenshots
![xo9n6ceyxo](https://user-images.githubusercontent.com/1103543/49970669-f0bf5300-fee0-11e8-8292-f02f828c1cc8.gif)



## Related PRs
 
<!-- Add any related PR for these changes. -->

- Type Settings PR: N/A <!-- https://github.com/foxbroadcasting/fng-cms-settings/pull/{PR#} or https://github.com/foxbroadcasting/fng-cms-settings/compare/dev...{settings branch name} -->
- API PR: N/A <!-- https://github.com/foxbroadcasting/dcgapi-services/pull/{PR#} -->

## Checklist

<!-- Verify that you have done all of the following and mark them as done. -->

- [ ] I added the necessary documentation, if appropriate.
- [ ] I added tests to prove that my fix is effective or my feature works.
- [ ] I reviewed existing Pull Requests before submitting mine.

Commit:
chore: DPP-1231 Add chevron icon and update styles

Commit:
chore: DPP-1231 Remove up chevron icon

Commit:
chore: DPP-1231 Update styles for better UX

Commit:
chore: DPP-1231 Remove parameter in the arrow render

Commit:
chore: DPP-1231 Update tab identation

ragrum commented:
@sabs231 when modifying the `package.json` as mentioned, I get this error:

```
npm ERR! Command failed: /usr/local/bin/git checkout bug/DPP-1231
npm ERR! error: pathspec 'bug/DPP-1231' did not match any file(s) known to git
npm ERR!
```

ragrum commented:
@sabs231 I found why, this:

```
"@foxdcg/react-select": "git://github.com/foxbroadcasting/react-select.git#bug/DPP-1231",
```

should be instead this:

```
"@foxdcg/react-select": "git://github.com/foxbroadcasting/react-select.git#chore/DPP-1231",
```

ragrum reviewed (changes requested):
In the mockups there's no grey area around the dropdown menu, the search results should align to the right with the drop-down menu, and it seems there's still missing the padding at the bottom.

![image](https://user-images.githubusercontent.com/124746/48851942-ea223d80-ed71-11e8-98df-eafe4e90fb2a.png)

sabs231 commented:
@ragrum Yes, is `chore/DPP-1231` forgot to updated it there

ragrum commented:
As for the "grey area" mentioned above, I already talked about this with Sabs, and that happens with any other field being filled in, so never mind about that one.

Commit:
chore: DPP-1231 Update styles according to comments

ragrum reviewed (approved):
After styles correction, this works as described.

omar-aguilar commented on less/select.less:
```diff
@@ -22,8 +22,8 @@
 @select-input-border-width:         1px;
 @select-input-height:               36px;
 @select-input-internal-height:      (@select-input-height - (@select-input-border-width * 2));
-@select-input-placeholder:          #aaa;
```
If you need a `000` color, please make it only for the unique select, this changes affects the other select items as well

| Before | After |
| --- | --- |
| ![screen shot 2018-11-26 at 10 36 32 am](https://user-images.githubusercontent.com/1847316/49028186-69938080-f167-11e8-908b-39642fe0e301.png) | ![screen shot 2018-11-26 at 10 36 28 am](https://user-images.githubusercontent.com/1847316/49028187-69938080-f167-11e8-8a38-8e75390538a8.png) |

omar-aguilar commented on less/select.less:
```diff
@@ -49,7 +49,7 @@
 
 // clear "x" button
 @select-clear-size:                 floor((@select-input-height / 2));
-@select-clear-color:                #999;
```
Please make it only for the unique input, as this affects to all inputs

| Before | After |
| --- | --- |
| ![screen shot 2018-11-26 at 10 41 22 am](https://user-images.githubusercontent.com/1847316/49028437-07874b00-f168-11e8-97df-d599bf6ff00e.png) |  ![screen shot 2018-11-26 at 10 41 33 am](https://user-images.githubusercontent.com/1847316/49028435-07874b00-f168-11e8-8865-0763edb5d121.png) |

sabs231 replied:
Yeah, I talked to UX team about that and they are expecting to do that change on all of the selects

omar-aguilar commented on less/select.less:
```diff
@@ -22,8 +22,8 @@
 @select-input-height:               36px;
 @select-input-internal-height:      (@select-input-height - (@select-input-border-width * 2));
-@select-input-placeholder:          #aaa;
-@select-text-color:                 #333;
```
Please make it only for the unique input, as this affects to all inputs
After this the text looks more darker

| Before | After |
| --- | --- |
| ![screen shot 2018-11-26 at 10 41 22 am](https://user-images.githubusercontent.com/1847316/49028437-07874b00-f168-11e8-97df-d599bf6ff00e.png) |  ![screen shot 2018-11-26 at 10 41 33 am](https://user-images.githubusercontent.com/1847316/49028435-07874b00-f168-11e8-8865-0763edb5d121.png) |

sabs231 replied:
Yeah, I talked to UX team about that and they are expecting to do that change on all of the selects

omar-aguilar commented on src/Select.js:
```diff
@@ -1325,7 +1325,7 @@ Select.defaultProps = {
 	optionComponent: Option,
 	pageSize: 5,
-	placeholder: 'Select...',
+	placeholder: 'Select',
```
Please change the placeholder only for you unique input, this affects the other elements as well

| Before | After |
| --- | --- |
| ![screen shot 2018-11-26 at 10 46 37 am](https://user-images.githubusercontent.com/1847316/49028756-c17eb700-f168-11e8-9417-b6e356b659a1.png) | ![screen shot 2018-11-26 at 10 46 42 am](https://user-images.githubusercontent.com/1847316/49028688-985e2680-f168-11e8-8fce-190d9401df4f.png) |

sabs231 replied:
Yeah, I talked to UX team about that and they are expecting to do that change on all of the selects

omar-aguilar commented on src/Async.js:
```diff
@@ -40,7 +40,7 @@ const defaultProps = {
 	ignoreAccents: true,
 	ignoreCase: true,
-	loadingPlaceholder: 'Loading...',
+	loadingPlaceholder: '',
```
This legend appears when the options are being loaded, or for example when you close a reference list,
please remove it only for the unique selector

sabs231 replied:
Yeah, I talked to UX team about that and they are expecting to do that change on all of the selects

omar-aguilar commented on src/utils/defaultArrowRenderer.js:
```diff
@@ -4,9 +4,12 @@ import PropTypes from 'prop-types';
 			onMouseDown={onMouseDown}
-		/>
+		>
+			<svg viewBox="0 0 24 24" preserveAspectRatio="xMidYMid meet">
```
That arrow will need to be centered
![screen shot 2018-11-26 at 10 46 42 am](https://user-images.githubusercontent.com/1847316/49028939-233f2100-f169-11e8-93c2-25ec985bf7a4.png)

sabs231 replied:
do you mean vertically centered?

omar-aguilar replied:
yes

omar-aguilar reviewed (changes requested):
Checking at the zeplin mock https://app.zeplin.io/project/5a3750c7e8354b6a0e200872/screen/5bef3ee81812d23ec8a1b1df

* Looks like there is an empty state that says  `No Items Selected`
    <img width="256" alt="screen shot 2018-11-26 at 10 22 11 am" src="https://user-images.githubusercontent.com/1847316/49027198-4798fe80-f165-11e8-82ec-c56aace75699.png">
* There is a placeholder in the search input that says `Search`
    ![screen shot 2018-11-26 at 10 52 16 am](https://user-images.githubusercontent.com/1847316/49029040-5f728180-f169-11e8-9dc3-665ab722d5a7.png)
* The color for both legends above is `rgba(0, 0, 0, 0.38)`

Commit:
chore: DPP-1231 Make parent fit the size of content

Commit:
chore: DPP-1231 Remove little unused styles

Commit:
chore: DPP-1231 Add color when state changes

Commit:
chore: DPP-1231 Update loading logic for outer menu

Commit:
chore: DPP-1231 Fix search icon styles

Commit:
chore: DPP-1231 Add rest of styles

Commit:
chore: DPP-1231 Revert back other select changes

Commit:
chore: DPP-1231 Fix loading placeholder text color

omar-aguilar commented:
Can we remove the background of the FieldFrame that contains the panel selector 

| Current | Mock |
| --- | --- |
| ![screen shot 2018-12-17 at 2 12 16 pm](https://user-images.githubusercontent.com/1847316/50112827-dac5d100-0205-11e9-9689-dbbd0f72cd90.png) | ![screen shot 2018-12-17 at 2 13 24 pm](https://user-images.githubusercontent.com/1847316/50112892-05b02500-0206-11e9-80d7-eda667d59ad0.png) |

omar-aguilar commented:
#### There are some missing changes according to the mocks
1. The cursor when hovering the search box should not be the default one
1. The max width should be 168px and the text should be truncated
    ![screen shot 2018-12-17 at 1 51 01 pm](https://user-images.githubusercontent.com/1847316/50111664-e2d04180-0202-11e9-9595-19195ab7e230.png)
1. background on option hover should be `rgba(0, 0, 0, 0.04)`
    ![screen shot 2018-12-17 at 1 48 51 pm](https://user-images.githubusercontent.com/1847316/50111557-a56bb400-0202-11e9-9a7a-f7cd403eb6f4.png)
1. The border when hover should not change

    | Current | Mock |
    | --- | --- |
    |  ![screen shot 2018-12-17 at 2 09 10 pm](https://user-images.githubusercontent.com/1847316/50112738-ab16c900-0205-11e9-83f0-eeaf7317a0d8.png) | ![screen shot 2018-12-17 at 2 11 27 pm](https://user-images.githubusercontent.com/1847316/50112752-aeaa5000-0205-11e9-93e1-41194d0c431b.png) |
1. The search placeholder doesn't appear 

    | Current | Mock |
    | --- | --- |
    | ![screen shot 2018-12-17 at 1 59 32 pm](https://user-images.githubusercontent.com/1847316/50112122-1069ba80-0204-11e9-862c-9d84d2537190.png) | ![screen shot 2018-12-17 at 1 59 44 pm](https://user-images.githubusercontent.com/1847316/50112132-16f83200-0204-11e9-92be-e600976f0e2e.png) |

#### The Assign to Panel label gets lost when no option is selected

| Case 1 | Case 2 | Case 3 |
| --- | --- | --- |
| ![screen shot 2018-12-17 at 1 40 54 pm](https://user-images.githubusercontent.com/1847316/50111558-a56bb400-0202-11e9-86aa-3bf0e72e72b7.png) | ![screen shot 2018-12-17 at 1 40 25 pm](https://user-images.githubusercontent.com/1847316/50111559-a56bb400-0202-11e9-9dae-872cbc5d5267.png) | ![screen shot 2018-12-17 at 1 40 36 pm](https://user-images.githubusercontent.com/1847316/50111560-a56bb400-0202-11e9-9828-d5176205b908.png) |


#### The Clear icon appears until the content is loaded 
![screen shot 2018-12-17 at 1 56 36 pm](https://user-images.githubusercontent.com/1847316/50111998-c385e400-0203-11e9-8c72-483e1ea30820.png)

Commit:
chore: DPP-1231 Update styles from comments

omar-aguilar commented on src/Select.js:
```diff
@@ -749,11 +749,14 @@ class Select extends React.Component {
 			? (<div className="Select-unique-placeholder">No Item Selected</div>)
 			: (<div className="Select-placeholder">{this.props.placeholder}</div>);
 		if (!valueArray.length) {
+      if (isTop) {
```
there is some code misalignment

omar-aguilar commented on src/Select.js:
```diff
@@ -871,13 +874,14 @@ class Select extends React.Component {
 		return (
 			<div className={ className } key="input-wrap">
-				<input {...inputProps} />
+				<input  {...inputProps} />
```
there is an extra space

omar-aguilar commented on src/Select.js:
```diff
@@ -1132,9 +1136,9 @@ class Select extends React.Component {
-						<div className="Select-multi-value-wrapper" id={this._instancePrefix + '-value'}>
-							{this.renderInput(valueArray, focusedOptionIndex)}
-						</div>
+							<div className="Select-multi-value-wrapper" id={this._instancePrefix + '-value'}>
```
looks like there is an extra tab in this part of code

omar-aguilar reviewed (commented):
1. Looks like the hover background doesn't cover the whole line
    <img width="332" alt="screen shot 2019-01-02 at 11 27 49 am" src="https://user-images.githubusercontent.com/1847316/50606652-badc1180-0e83-11e9-8353-cb6ad223e1df.png">
1. The search icon color should be `rgba(0, 0, 0, 0.54)`
    <img width="194" alt="screen shot 2019-01-02 at 11 33 19 am" src="https://user-images.githubusercontent.com/1847316/50606653-badc1180-0e83-11e9-80db-b64e31bed7db.png">
1.  With the change of the remove button displayed all the time there is an issue for example when you go to http://local.dpp.foxdcg.com:3000/new/videolists?filter=manual and select a SEO and then close the dropdown, both the loading an clear buttons appear at the same time, before only one icon was shown at a time
    <img width="90" alt="screen shot 2019-01-02 at 11 38 38 am" src="https://user-images.githubusercontent.com/1847316/50606654-bb74a800-0e83-11e9-8a85-7b04991dc54e.png">
1. Similar to the above, when there is no multi select, there is a weird padding/margin at the bottom and right
   <img width="707" alt="screen shot 2019-01-02 at 11 27 57 am" src="https://user-images.githubusercontent.com/1847316/50606655-bb74a800-0e83-11e9-8131-20ddf3e25347.png">

Commit:
chore: DPP-1231 Fix minor style issues

- Hover option background does not cover the whole line
- Search icon updated with correct fill color
- Fix style mixes between unique select and single select

Commit:
chore: DPP-1231 Update to next version
